sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -1170,4 +898,44 @@ private void waitForFuturesToFinish() throws ExecutionException, InterruptedExce
       }
     }
   }
+
+  /**
+   * A {@link PTransform} to read data from Apache Cassandra. See {@link CassandraIO} for more
+   * information on usage and configuration.
+   */
+  @AutoValue
+  public abstract static class ReadAll<T> extends PTransform<PCollection<Read<T>>, PCollection<T>> {
+
+    @Nullable
+    abstract Coder<T> coder();
+
+    abstract ReadAll.Builder<T> builder();
+
+    /** Specify the {@link Coder} used to serialize the entity in the {@link PCollection}. */
+    public ReadAll<T> withCoder(Coder<T> coder) {
```

Comment:
I am not sure if we need this, I assume the coder comes form the `Read<T> input.

Reply:
I think we do, but am not sure.  You have to call `setCoder` on the PCollection itself, so we don't have access to the `Read<T>` at a point when we also have access to a single `Read<T>` (they are only supplied in the `ReadFn` which can't call `setCoder` on the returned PTransform).  Is my thinking correct?  I could be unaware of a different way to do this.